grype/version/fuzzy_constraint_test.go

```diff
@@ -59,6 +59,10 @@ func TestFuzzyVersionComparison(t *testing.T) {
 		{"8.0.456", "8.0.457", -1},
 		{"8.0.456+1", "8.0.456+2", -1},
 		{"8.0.456", "8.0.456+1", -1},
+		// Test case for fuzzy version comparison bug with patch numbers
+		// This should pass: 4.2.8p9 < 4.2.8p15 (p9 comes before p15 numerically)
+		// But currently fails due to lexicographic fallback where "p9" > "p15" (string comparison)
+		{"4.2.8p9", "4.2.8p15", -1},
```

Comment:
Did I read in the discussion that openssl uses the fuzzy matcher; do you happen to know if that's accurate? If so, are any of these version strings testing openssl versions? if not, that might be the only suggestion I have for improving the coverage here to give me a warmer fuzzy, for example I see [openssl 1.1.1p](https://github.com/openssl/openssl/releases/tag/OpenSSL_1_1_1p)

Reply:
@kzantow I've added test cases for some old openssl versions.